lib/jit_preloader/active_record/associations/singular_association.rb

```diff
@@ -4,7 +4,7 @@ module ActiveRecordAssociationsSingularAssociation
     def load_target
       was_loaded = loaded?
 
-      if !loaded? && owner.persisted? && owner.jit_preloader
+      if !loaded? && owner.persisted? && owner.jit_preloader && (reflection.scope.nil? || reflection.scope.arity == 0)
```

Comment:
Note that we already have this guard on our patches for the `preload` (Rails <= 5.1) and `run` (Rails >= 5.2) methods. Is it needed here as well?

Reply:
I believe the first side will help deal with things when you are initially preloading, but you can see from the spec, that even with those previous changes in place, without this change, accessing an association that has an instance dependent scope will cause it to throw an exception.